pointer/pointer.go

```diff
@@ -47,57 +47,20 @@ func AllPtrFieldsNil(obj interface{}) bool {
 	return true
 }
 
-// Int returns a pointer to an int
-func Int(i int) *int {
-	return &i
+// Ref returns a pointer to the value.
+func Ref[T any](v T) *T {
```

Comment:
Why not "New"  `ptr.New[int64](12345)` is as close as we can get to `&int64(12345)` or `new(int64, 12345)`

Reply:
I liked how complementary `Ref` and `Deref` sounded. But `New` would work too.

Reply:
How about `To`? `pointer.To(12345)` seems even nicer to me...

Reply:
`pointer.New()`, `pointer.To()`, `pointer.P()` would all be fine to me.  We could even rename the pkg to `ptr`.

Reply:
can we? I guess we would need to duplicate the package first.

But we are still waiting for 1.24 to go EoL next month.

Reply:
It wouldn’t have to be duplicated — you could add a generic `ptr` package, and change `pointer` to be full of vars pointing to `ptr` functions.

Reply:
One benefit of introducing a new generic `ptr` package instead of updating the existing `pointer` package would be to simplify migration tracking: any references to `utils/pointer` would point (hah) to code that should be updated.

Reply:
I agree.

@skitt would you like to take over? The time for adding these generics is coming next month, and I honestly have limited time to do so.

I can simply close this PR and you can start over with the `ptr` package approach.

Reply:
I can take care of this, yes!